subsys/net/lib/nrf_cloud/src/nrf_cloud_pgps.c

```diff
@@ -1172,6 +1192,16 @@ static int open_storage(uint32_t offset, bool preserve)
 	/* assume cache is no longer valid */
 	discard_prediction_buffer();
 
+#if defined(CONFIG_PM_PARTITION_REGION_PGPS_EXTERNAL)
+	err = k_sem_take(&flash_accessible, K_NO_WAIT);
```

Comment:
if this function (`open_storage`) returns successfully, then the corresponding `k_sem_give` is in `consume_pgps_data` correct?  

is there any case where `consume_pgps_data` could not get called?

Reply:
Correct.

And good point.